esphome/components/cap1188/cap1188.cpp

```diff
@@ -0,0 +1,88 @@
+#include "cap1188.h"
+#include "esphome/core/log.h"
+#include "esphome/core/hal.h"
+
+namespace esphome {
+namespace cap1188 {
+
+static const char *const TAG = "cap1188";
+
+void CAP1188Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up CAP1188...");
+
+  // Reset device using the reset pin
+  if (this->reset_pin_ != nullptr) {
+    this->reset_pin_->pin_mode(esphome::gpio::FLAG_OUTPUT);
```

Comment:
You should setup the pin instead, the code generator takes care of setting the output mode.
```suggestion
    this->reset_pin_->setup();
```